### PR TITLE
Scss bootsrap 4

### PR DESCRIPTION
**Pull request by ger-benjamin**
GSGMF-418

See also: https://getbootstrap.com/docs/4.0/migration/

### CODE REVIEW

File: contribs/gmf/src/sass/vars_only.scss [hunk 1]
```diff
@@ -34,13 +34,19 @@ $above-all: 9999;
 
 $nav-width: 20rem !default;
 
-$border-radius-base: 2px !default;
-$border-radius-small: 1px !default;
+$border-radius-base: 0 !default;
+$border-radius-small: 0 !default;
```
Comment (sbrunner) [inline contribs/gmf/src/sass/vars_only.scss]: Shouldn't we remove them and use `$border-radius-sm` and `$border-radius`

File: contribs/gmf/src/sass/vars_only.scss [hunk 2]
```diff
@@ -34,13 +34,19 @@ $above-all: 9999;
...
 $input-border-focus: darken($brand-primary, $standard-variation) !default;
 
 $fa-font-path: "~font-awesome/fonts";
 
 // Bootstrap 3 compatibility layer
+$border-radius: $border-radius-base !default;
+$border-radius-lg: $border-radius-base !default;
+$border-radius-sm: $border-radius-base !default;
```
Comment (sbrunner) [inline contribs/gmf/src/sass/vars_only.scss]: shouldn't be `$border-radius-base / 2 !default`?

File: contribs/gmf/src/controllers/desktop.scss [hunk 3]
```diff
@@ -247,7 +246,7 @@ gmf-backgroundlayerselector {
 $theme-selector-columns: 2;
 .gmf-theme-selector li {
   float: left;
-  width: calc((100% - $theme-selector-columns * 2 * $half-app-margin) / $theme-selector-columns);
+  width: calc((100% - #{$theme-selector-columns} * 2 * #{$half-app-margin}) / #{$theme-selector-columns});
```
Comment (sbrunner) [inline contribs/gmf/src/controllers/desktop.scss]: What's this syntax?

File: contribs/gmf/src/sass/vars_only.scss (hunk 1 above)
Comment (ger-benjamin) [reply contribs/gmf/src/sass/vars_only.scss]: +1

File: contribs/gmf/src/controllers/desktop.scss (hunk 3 above)
Comment (ger-benjamin) [reply contribs/gmf/src/controllers/desktop.scss]: The equivalent to calc(100% - ~"@{theme-selector-columns} * 2 ...") in less

So a manner to use `calc` with sass variables

File: contribs/gmf/src/sass/vars_only.scss (hunk 1 above)
Comment (ger-benjamin) [reply contribs/gmf/src/sass/vars_only.scss]: In fact, I prefer to use our own variable ($border-radius-base) So If customers want a radius with bootstrap thing they can have it. And using our own variable, we are not dependant of boostrap as well.

We never use $border-radius-small, I remove it.

Comment (sbrunner) [reply contribs/gmf/src/sass/vars_only.scss]: OK :-)

File: contribs/gmf/src/sass/vars_only.scss (hunk 2 above)
Comment (sbrunner) [reply contribs/gmf/src/sass/vars_only.scss]: OK :-)

File: contribs/gmf/src/sass/vars_only.scss [hunk 4]
```diff
@@ -34,13 +34,18 @@ $above-all: 9999;
...
+$font-size-base: 0.8rem !default;
+$input-btn-padding-y: 0.25rem !default;
+$input-btn-padding-x: 0.5rem !default;
```
Comment (sbrunner) [inline contribs/gmf/src/sass/vars_only.scss]: Shouldn't we set the comment
`# Bootstrap 4 customization`
for this block?

Comment (ger-benjamin) [reply contribs/gmf/src/sass/vars_only.scss]: You mean you want that I change the `// Bootstrap 3 compatibility layer` comment to yours ?

**Comment by ger-benjamin** [comment]
Still a lot to do (mainly in the right panels) but I want to merge
It's already far better now :-)

**Pull request merged**